core/src/main/java/org/jsmart/zerocode/core/engine/preprocessor/ZeroCodeExternalFileProcessorImpl.java

```diff
@@ -147,6 +155,21 @@ void digReplaceContent(Map<String, Object> map) {
                             throw new RuntimeException(exx);
                         }
                     }
+                    if (token != null && token.startsWith(YML_PAYLOAD_FILE)) {
+                        String resourceJsonFile = token.substring(YML_PAYLOAD_FILE.length());
+                        try {
+                            JsonNode jsonNode = yamlMapper.readTree(readYamlAsString(resourceJsonFile));
+                            if (jsonNode.isObject()) {
+                                final Map<String, Object> yamlFileContent = objectMapper.convertValue(jsonNode, Map.class);
+                                digReplaceContent(yamlFileContent);
+                                jsonNode = objectMapper.convertValue(yamlFileContent, JsonNode.class);
+                            }
+                            entry.setValue(jsonNode);
+                        } catch (Exception exx) {
+                            LOGGER.error("External file reference exception - {}", exx.getMessage());
+                            throw new RuntimeException(exx);
+                        }
+                    }
```

Comment:
this added bloc have some duplicated lines with the JSON management below, could you refactor the entire code block (from line 130 to line 172) in order to reuse the common implementation for both JSON and YAML ?

You can extract the JsonNode creation in top of the algorithm like 
```
                            JsonNode jsonNode = token.startsWith(YML_PAYLOAD_FILE) ? yamlMapper.readTree(readYamlAsString(token.substring(YML_PAYLOAD_FILE.length()))) : objectMapper.readTree(readJsonAsString(token.substring(JSON_PAYLOAD_FILE.length())));

```

With this approach,  the rest of code remains untouched